### ISSUE TITLE
New ramp quest needs better description (separate mapped ramp) 

### ISSUE THREAD

**Issue by RubenKelevra** [westnordost/StreetComplete#2168]
Thanks for all the cool new quests in the last weeks, I really appreciate it! 

The ramp quest however feels a bit unfinished. Since steps are usually a lot steeper than a wheelchair ramp can be, the wheelchair ramps tend to be separate and not directly attached to the stairs.[0]

This often leads to the wheelchair ramp being mapped as a separate way. 

Currently there's a no information for the users that in the case that there's a separately mapped ramp (shown as a regular way on the map) they should select 'no' to avoid that the stairs itself are considered passable by wheelchairs. 

We have a similar situation with pavements along of roads, which gets mapped separately. On this quest there's an option to map them separately[1].

I see two possible solutions:

- Show a pop-up on each 'there's a wheelchair ramp' answer to the user, that he/she should check for a separately mapped way and in case it's mapped separately select 'no' instead or unselect the wheelchair option. 
- Add a fifth option which conflicts with the wheelchair selection and sets a value like 'separate' for the wheelchair ramp in the osm data.

**Picture 0**

![Screenshot_20201016-002109](https://user-images.githubusercontent.com/614929/96193072-a7f6dc00-0f47-11eb-8689-36e112d79a49.jpg)

**Picture 1**

![Screenshot_20201016-002934](https://user-images.githubusercontent.com/614929/96193098-b80ebb80-0f47-11eb-976a-dd7b4b0de544.jpg)

**Versions affected**
25 beta 1

**Comment by peternewman** [comment]
Yeah I was going to raise this too, there's even an equivalent tag:
https://wiki.openstreetmap.org/wiki/Key:ramp#Ramps

`ramp=separate`

**Comment by RubenKelevra** [comment]
> Yeah I was going to raise this too, there's even an equivalent tag:
> https://wiki.openstreetmap.org/wiki/Key:ramp#Ramps
> 
> `ramp=separate`

Well, I think there are still steps which got built with say a bicycle ramp but got a ramp for wheelchairs added. 

So I think we should use `ramp:wheelchair=separate` instead - since bicycles might be omitted from using the wheelchair ramp for example.

**Comment by peternewman** [comment]
> So I think we should use `ramp:wheelchair=separate` instead - since bicycles might be omitted from using the wheelchair ramp for example.

Although there are also places with some normal plain stairs (say three or four steps) and a ramp running straight down in parallel, it would seem incorrect to me to use the wheelchair separate as that suggests it's a wheelchair only ramp to me, or the others can't use it, in reality there are no such signed restrictions.

Also 291 `ramp=separate` versus 7 tags for `ramp:wheelchair=separate`.

**Comment by rhhsm** [comment]
I think the image for the scroller ramp can be improved. I think in this image, both the lower and the upper ramp are scroller ramps (with/without steps between the tracks).
![image](https://user-images.githubusercontent.com/69100427/96264066-7a07ab00-0fcc-11eb-94e8-c0590a36c133.png)

**Comment by peternewman** [comment]
Just to say having done it in the real world, :100: % on the separate ramp, I had to leave notes for five or six and fix with local edits, there weren't any I mapped that had a ramp that wasn't separate!

**Comment by RubenKelevra** [comment]
> > So I think we should use `ramp:wheelchair=separate` instead - since bicycles might be omitted from using the wheelchair ramp for example.
> 
> Although there are also places with some normal plain stairs (say three or four steps) and a ramp running straight down in parallel, it would seem incorrect to me to use the wheelchair separate as that suggests it's a wheelchair only ramp to me, or the others can't use it, in reality there are no such signed restrictions.

Point is, `ramp=separate` isn't specific enough. Is the separate ramp now accessible for wheelchairs or not? 🤔 

There's currently no way to find the 'matching' ramp-way for stairs. So stairs must be considered as unpassable for wheelchairs, except there's explicitly a wheelchair ramp tagged. 

When we add `ramp=separate` nothing has changed - the router still have no clue if it's safe to use the ramp with a wheelchair.

While a router can just route 'through'  the steps itself, if there's a ramp for wheelchairs tagged, since the wheelchair user will know which way exactly is appropriate to use. 

> Also 291 `ramp=separate` versus 7 tags for `ramp:wheelchair=separate`.

Yeah, that's expected, since when a user created the ramp, they will just add a separate way next to the steps and call it a day. 

The issue is, that there's no information on the way next to the steps if it's wide and flat enough for safe wheelchair use. 

> Just to say having done it in the real world, 💯 % on the separate ramp, I had to leave notes for five or six and fix with local edits, there weren't any I mapped that had a ramp that wasn't separate!

Yeah, I true for me too. All steps I've visited so far popping up in SC had separate ramps already mapped or no ramps at all. 

It's currently 8:1 separate / none.

I mean left all 'separate' quests open, but I fear that someone else might add wrong data with SC since there's no hint that separate ramps should be ignored.

**Comment by matkoniecz** [comment]
> There's currently no way to find the 'matching' ramp-way for stairs. So stairs must be considered as unpassable for wheelchairs, except there's explicitly a wheelchair ramp tagged.

In general it is not necessary to find it.

> When we add ramp=separate nothing has changed - the router still have no clue if it's safe to use the ramp with a wheelchair.

Router may now treat steps geometry as not routable (no need to guess whatever there is ramp or not), as explicitly mapped ramp geometry will be used for routing.

In contrast, using `ramp:wheelchair=yes` may cause router to use steps as part of geometry for routing what is slightly worse.

See https://www.openstreetmap.org/#map=19/50.08170/19.85768&layers=C where it is preferable for wheelchair routing to use mapped separate ramp (it also would result in a better time estimate, routing over `highway=steps ramp:wheelchair=yes` would need guessing how long the ramp is).

![screen](https://user-images.githubusercontent.com/899988/96542906-706b9500-12a3-11eb-9422-9b0afe8314aa.jpg)


Also, mappers know that ramp is handled so there is no need to check it (for example with SC quest).

**Comment by peternewman** [comment]
> > > So I think we should use `ramp:wheelchair=separate` instead - since bicycles might be omitted from using the wheelchair ramp for example.

Perhaps it's a cultural thing but in the UK a ramp is a ramp, you could use it for a stroller/pushchair or an upright shopping trolley. The only thing I can possibly think of is it might be awkward for a bike to use it if it's a ramp with a few tight turns, but I don't think I've seen it not-permitted. Anyway because of those cases if we're going to tag the individual wheelchair case, surely we should do the same for the others so parents can (eventually) plan a route which is pushchair friendly?

Maybe my bike example is the only case where there might be a rail to run a bike down the steps, but everything else would use the ramp.

To take a similar case, a lift which goes say have a storey to avoid some steps wouldn't be exclusively for wheelchair users, although that might be the reason it had been put in; no-one would ever stop someone putting a pushchair in and make them carry it up the stairs (although I'll admit a bike may get strange looks if it's small).

> I left all 'separate' quests open, but I fear that someone else might add wrong data with SC since there's no hint that separate ramps should be ignored.

I've specifically tagged them as ramp=separate as that's what the wiki says:
https://wiki.openstreetmap.org/wiki/Key:ramp

> Separate ramp geometry would benefit from `wheelchair=no` if unsafe to use, but in general I am not aware about separately mappable ramps that would not be a wheelchair ramps

Exactly this, it seems quite rare to have a separate ramp which isn't wheelchair suitable (why was it built otherwise), and if so surely it should be tagged on that rather than on a related set of steps. You could also have a building which only had a ramp at the entrance but that ramp may be unsuitable for wheelchairs. Surely we should be working on finding and tagging these ramps rather than just ones related to stairs?

Are lifts/elevators covered/related, should we somehow be tagging them if they are the separate option to steps (I can't see anything in the wiki).

**Comment by matkoniecz** [comment]
>  it might be awkward for a bike to use it if it's a ramp with a few tight turns, but I don't think I've seen it not-permitted

https://www.openstreetmap.org/way/242699349

**Comment by RubenKelevra** [comment]
by @matkoniecz
> Separate ramp geometry would benefit from `wheelchair=no` if unsafe to use, but in general I am not aware about separately mappable ramps that would not be a wheelchair ramps

Well, there are sometimes ramps which are just too steep to climb them with a (manual) wheelchair. It's often just badly designed ones.

The ramp in the Wuppertal main train station is an example[1].

But I agree that this might be a pretty small edge-case.

[1] https://www.wz.de/nrw/wuppertal/pro-bahn-kritisiert-fehlende-funktionalitaet-am-wuppertaler-hauptbahnhof_aid-24985661

---

To sum it up:

We should have a dedicated button for 'seperate' ramps for wheelchairs which are already on the map. And maybe have a popup showing up when the user selects 'wheelchair ramp' and not the separate one, to notify him/her to check if the ramp is maybe mapped seperately.

The tag should either be `ramp=separate` which is quite general, or more specifically `ramp:wheelchair=separate`.

Maybe just vote on this comment what you think is better (tag-wise):

`ramp=separate` - 🎉
`ramp:wheelchair=separate` - 🚀

**Comment by peternewman** [comment]
> I have seen it, for example https://www.openstreetmap.org/way/242699349

Ah okay. I assume it's not the normal there though, just happens sometimes?

> The ramp in the Wuppertal main train station is an example[1].
> 
> But I agree that this might be a pretty small edge-case.

I didn't see it in the pictures, I assume it's in the German text?

> We should have a dedicated button for 'seperate' ramps for wheelchairs which are already on the map. And maybe have a popup showing up when the user selects 'wheelchair ramp' and not the separate one, to notify him/her to check if the ramp is maybe mapped seperately.

Surely the reverse here, if you select wheelchair then ideally it actually is wheelchair accessible (or worst case you've said the stairs are when actually it's a separate ramp, but routing for the user will still work), the pop-up should be when you select no ramp to ask if it's actually separate so you don't incorrectly tag the stairs as un-routable to a wheelchair when there is in fact a separate ramp which is suitable.

> The tag should either be `ramp=separate` which is quite general, or more specifically `ramp:wheelchair=separate`.

I'm happy with both, but I think we should either offer both or decide which is more common and have an automatic tag for that one and a leave a note for the other, or automatic for both. IMHO you still haven't addressed how someone handles `ramp:stroller=separate` if the ramp is stroller and wheelchair suitable which is very likely.

Or to put it another way, we've seen one that is presumably designated as no bikes from @matkoniecz , has anyone seen one designated as no strollers or no shopping trolleys?

In fact, thinking about this further, at least in the UK having a ramp of any sort parallel to the steps is pretty rare, even if you're only going down a few steps, similar to the picture, the ramp will normally be three or four times longer (unless they've deliberately made the steps extremely shallow as in the picture), at which case if they are different lengths shouldn't we be mapping them separately anyway. Also a lot of the time unless they are off a terraced area, the ramp has to zig-zag or take another different route to get sufficient length, in which case it should definitely be separate.

**Comment by matkoniecz** [comment]
> Ah okay. I assume it's not the normal there though, just happens sometimes?

Many ramps are explcitly forbidden or leading into places where cycling is pointless or illegal (for example noone will cycle through platform with 180 degree turns and lock bicycle next to the door, it is much easier to lock your bicycle and walk upstairs).

See https://www.openstreetmap.org/way/407673711#map=19/50.08096/19.88483 https://www.openstreetmap.org/way/407894794 https://www.openstreetmap.org/way/409822220

**Comment by westnordost** [comment]
When wheelchair ramp was selected, I will simply show a dialog that asks whether the ramp is displayed separately on the map. Something like:

> Is the wheelchair ramp for these steps displayed as a separate line on the map?
>
> [I will check] [separate] [not separate]

**Comment by RubenKelevra** [comment]
> > The ramp in the Wuppertal main train station is an example[1].
> > But I agree that this might be a pretty small edge-case.
> 
> I didn't see it in the pictures, I assume it's in the German text?

Yes. The ramp looks like it's made for wheelchairs, but is 3 times steeper - way too steep to get up and down safely. 

Additionally there's a blind corner right at the end, so you might hurt someone when you're driving down, since you cannot see anything. 

Had a hard time finding a picture, but at least the blind corner (on the left) is visible here:

![38382359965_e259a2a6aa_b](https://user-images.githubusercontent.com/614929/97105979-32b1a680-16bf-11eb-8b9b-78c43e217d91.jpg)


---

Thanks for the fix @westnordost :)
